js/web/components/Profile.js

```diff
@@ -0,0 +1,377 @@
+import { useEffect, useContext, useState, useMemo } from 'react'
+import Head from 'next/head'
+import dynamic from 'next/dynamic'
+import { Box, Typography } from '@mui/material'
+import { styled } from '@mui/system'
+import Hub from '@nina-protocol/nina-internal-sdk/esm/Hub'
+import Release from '@nina-protocol/nina-internal-sdk/esm/Release'
+import { truncateAddress } from '@nina-protocol/nina-internal-sdk/src/utils/truncateAddress'
+
+const Dots = dynamic(() => import('./Dots'))
+const TabHeader = dynamic(() => import('./TabHeader'))
+const ReusableTable = dynamic(() => import('./ReusableTable'))
+const Profile = ({ profilePubkey }) => {
+
+  const {
+    getUserCollectionAndPublished,
+    releaseState,
+    filterReleasesPublishedByUser,
+    filterReleasesList,
+  } = useContext(Release.Context)
+
+  const { getHubsForUser, filterHubsForUser, hubState } = useContext(
+    Hub.Context
+  )
+
+  const [profilePublishedReleases, setProfilePublishedReleases] =
+    useState(undefined)
+  const [profileCollectionReleases, setProfileCollectionReleases] =
+    useState(undefined)
+  const [profileHubs, setProfileHubs] = useState(undefined)
+  const [activeView, setActiveView] = useState(undefined)
+  const [profileCollectionIds, setProfileCollectionIds] = useState(undefined)
+  const [fetchedUser, setFetchedUser] = useState(false)
+  const [fetchedReleases, setFetchedReleases] = useState(false)
+  const [fetchedHubs, setFetchedHubs] = useState(false)
+  const [fetchedCollection, setFetchedCollection] = useState(false)
+  const [views, setViews] = useState([
+    { name: 'releases', playlist: undefined, visible: false },
+    { name: 'collection', playlist: undefined, visible: false },
+    { name: 'hubs', playlist: null, visible: false },
+  ])
+
+  const artistNames = useMemo(() => {
+    if (profilePublishedReleases?.length > 0) {
+      return [
+        ...new Set(
+          profilePublishedReleases?.map(
+            (release) => release.metadata.properties.artist
+          )
+        ),
+      ]
+    }
+  }, [profilePublishedReleases])
+
+  useEffect(() => {
+    const getUserData = async (profilePubkey) => {
+      await getHubsForUser(profilePubkey)
+      const [collectionIds, publishedIds] = await getUserCollectionAndPublished(
+        profilePubkey
+      )
+      setProfileCollectionIds(collectionIds)
+    }
+    if (profilePubkey) {
+      setFetchedUser(true)
+      getUserData(profilePubkey)
+    }
+  }, [profilePubkey])
+
+  useEffect(() => {
+    let viewIndex
+    let updatedView = views.slice()
+    if (profilePubkey) {
+      const releases = filterReleasesPublishedByUser(profilePubkey)
```

Comment:
can you add spaces between different profile/collection/hub logic here for readability plz